### PR TITLE
Roundcube: add button to taskmenu for mailu

### PR DESCRIPTION
**Pull request by ItzYanick**
## What type of PR?

enhancement

## What does this PR do?

Adds a button to the roundcube interface. This button gets you back to the admin interface. Now you can switch between both without any issues

### Related issue(s)
- Did not found any but was bothering me

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by ItzYanick** [comment]
I do not know if any of the Prerequisites is needed for such a small change so I will just leave them unticked for now. Can change later if needed tho.

**Comment by ItzYanick** [comment]
Build failed because of docs? My changes could not have done this. #2364 seems to have the same problem

File: webmails/roundcube/mailu.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+rcmail.addEventListener('init', function(evt) {
+      var btn = document.createElement('a');
+      btn.setAttribute("href", "/admin");
```
Comment (Diman0) [inline webmails/roundcube/mailu.js:3]: You make the assumption that admin is always available on endpoint /admin. This is incorrect. It is defined in environment variable WEB_ADMIN. The path should be derived from this env var.

**Review by Diman0 — CHANGES_REQUESTED** [review]
See in-line comment. 
Even though it is a small change, please add a (changelog)[https://mailu.io/master/contributors/workflow.html#changelog] entry.

**Comment by Diman0** [comment]
bors try

**Comment by ItzYanick** [comment]
added changelog

**Comment by ItzYanick** [comment]
bors try

**Comment by Diman0** [comment]
bors try

File: webmails/roundcube/mailu.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+rcmail.addEventListener('init', function(evt) {
+      var btn = document.createElement('a');
+      btn.setAttribute("href", "{{ WEB_ADMIN }}");
+      btn.setAttribute("role", "button");
+      btn.setAttribute("class", "showurl");
+      btn.innerHTML = '<span class="inner">To Mailu</span>';
```
Comment (Diman0) [inline webmails/roundcube/mailu.js:6]: Please change this to
btn.innerHTML = '<span class="inner">To {{ SITENAME }}</span>';

Not everyone uses Mailu as the main site name. In SITENAME in mailu.env you can configure the name for the site. By default this is Mailu, but that is not the case for all users. Some users do not want Mailu branding at all.

**Review by Diman0 — CHANGES_REQUESTED** [review]
The PR itself works fine. 
See in-line comment. I did not think before about this. Not all users use Mailu as sitename. The value of SITENAME should be used in case someone does not use the Mailu branding.

There is also another topic. Multiple (or maybe even many) users have admin NOT exposed to the internet. For all these users we are introducing a broken 'back to the admin page' button.

I'm not certain if we should make this button optional. It would mean adding YET another setting to mailu.env. We have already to many settings here. So should all these users deal with it? They have a broken back to mailu button. Or do we add yet another config setting to mailu.env? I will ask the opinion of other people in the mailu dev channel and will come back to you.

**Comment by Diman0** [comment]
`There is also another topic. Multiple (or maybe even many) users have admin NOT exposed to the internet. For all these users we are introducing a broken 'back to the admin page' button..`

Ignore this. If the administrator blocked access, then it is the administrators problem. 
The default in Mailu is that admin is reachable. If admin is enabled, then the button should be enabled.  if admin is disabled, then the button should not be shown. The mailu.env setting is `ADMIN` for this. It is TRUE when admin is enabled, and FALSE when admin is disabled.

That means there should still be an IF statement around the button which checks for {{ ADMIN  }}.

**Comment by ghostwheel42** [comment]
> Ignore this. If the administrator blocked access, then it is the administrators problem.

Adding `if ({{ ADMIN }})` in front of the `rcmail.addEventListener()` in mailu.js would hide the button, wouldn't it?

**Comment by Diman0** [comment]
> > Ignore this. If the administrator blocked access, then it is the administrators problem.
> 
> Adding `if ({{ ADMIN }})` in front of the `rcmail.addEventListener()` in mailu.js would hide the button, wouldn't it?

It would hide the button when admin is disabled. So in the setup.mailu.io you disabled admin access. In this case it write ADMIN=FALSE to the mailu.env file.
ADMIN is true in mailu.env when you enabled Admin in setup.mailu.io. 

The situation I was talking about (the ignore this part), is when users block access to admin. E.g. make it only reachable via the local network and not via the external (WAN) network.

**Comment by ghostwheel42** [comment]
> The situation I was talking about (the ignore this part), is when users block access to admin. E.g. make it only reachable via the local network and not via the external (WAN) network.

Ah, okay. The button would only work on the "local network" then.
Admins can deceide themselves what to do when accessing the admin url from outside (display a message, redirect, etc.).

**Comment by Diman0** [comment]
It looks good to me. But I will first run some tests with admin disabled/enabled and a non-standard site name.
There are always many different scenario's to consider. Even with a small PR like this :grin: 

bors try

**Review by Diman0 — APPROVED** [review]
Tests went fine. LGTM. Thank you for your contribution.

**Comment by ghostwheel42** [comment]
> Tests went fine. LGTM. Thank you for your contribution.

It does what it should, but I think "To SITENAME" is not really the correct description for the button and below "settings" it's not the correct place. Also the button is only showing up in the Elastic skin.

Adding a section "Server" to the list of settings seems to be a better place.

I'm not against adding this, but I think using the section and opening the admin interface in a new tab (or don't open webmail in a new tab when coming from admin) would be a better way.

**Comment by ItzYanick** [comment]
Maybe the "To Admin" would be alright then because it also shows "Sign in Admin" at the login screen as well.

I don't want to have it in the Settings because I want a quick and easy way to switch. In the Admin Panel you can also click on "Webmail" to quickly get there so why not in the webmail screen as well. I have to re-investigate that if there is a plugin API of Roundcube that allows me adding buttons without messing with the UI. If there is not, I could offer to add the button on the left side *only* in the Elastic skin and in the other skins maybe in the settings.

I also think that the Admin Panel should open the webmail in the same tab instead of a own. This way we have a way more clean approach between both panels. But that would be a different Pull Request.

**Comment by ghostwheel42** [comment]
I've replaced this by #2591

**Pull request closed without merging**